File: app/src/main/java/me/wcy/music/fragment/BaseFragment.java
package me.wcy.music.fragment;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.hwangjr.rxbus.RxBus;

import me.wcy.music.activity.BaseActivity;
import me.wcy.music.service.PlayService;
import me.wcy.music.utils.PermissionReq;
import me.wcy.music.utils.binding.ViewBinder;

/**
 * 基类<br>
 * Created by wcy on 2015/11/26.
 */
public abstract class BaseFragment extends Fragment {
    protected Handler mHandler;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mHandler = new Handler(Looper.getMainLooper());
        ViewBinder.bind(this, getView());
        RxBus.get().register(this);
    }

    @Override
    public void onStart() {
        super.onStart();
        setListener();
    }

    protected void setListener() {
    }

    protected PlayService getPlayService() {
        BaseActivity activity = (BaseActivity) getActivity();
        return activity.playService;
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        PermissionReq.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    public void onDestroy() {
        RxBus.get().unregister(this);
        super.onDestroy();
    }
}

File: app/src/main/java/me/wcy/music/fragment/BaseFragment.kt
package me.wcy.music.fragment

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.fragment.app.Fragment
import com.hwangjr.rxbus.RxBus
import me.wcy.music.utils.binding.ViewBinder

/**
 * 基类<br></br>
 * Created by wcy on 2015/11/26.
 */
abstract class BaseFragment : Fragment() {
    protected var handler: Handler? = null
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        handler = Handler(Looper.getMainLooper())
        ViewBinder.bind(this, view)
        RxBus.get().register(this)
    }

    override fun onStart() {
        super.onStart()
        setListener()
    }

    protected open fun setListener() {}

    override fun onDestroy() {
        RxBus.get().unregister(this)
        super.onDestroy()
    }
}